planner/core/rule_join_reorder.go

```diff
@@ -91,10 +91,15 @@ func (s *joinReOrderSolver) optimizeRecursive(ctx sessionctx.Context, p LogicalP
 			return nil, err
 		}
 		schemaChanged := false
-		for i, col := range p.Schema().Columns {
-			if !col.Equal(nil, originalSchema.Columns[i]) {
-				schemaChanged = true
-				break
+		// We compare the length of the column in the schema first.
```

Comment:
Could you explain with the added case to show which column is added in the final schema?

Reply:
PTAL @Reminiscent